litgpt/api.py

```diff
@@ -406,12 +410,28 @@ def iterator():
                 include_prompt=False,
             )
 
-        if stream:
-            return outputs
-        elif return_as_token_ids:
-            return outputs
+        if benchmark:
+            benchmark_dict["Seconds total"] = time.perf_counter() - t0
+            benchmark_dict["Tokens generated"] = outputs.size(0) - input_ids.size(0)
+            benchmark_dict["Inference speed in tokens/sec"] = benchmark_dict["Tokens generated"] / benchmark_dict["Seconds total"]
+            if self.fabric.device.type == "cuda":
+                benchmark_dict["GPU memory used in GB"] = torch.cuda.max_memory_allocated() / 1e9
+
+            main_output = None
+            if stream:
+                main_output = outputs
+            elif return_as_token_ids:
+                main_output = outputs
+            else:
+                main_output = self.preprocessor.tokenizer.decode(outputs)
+            return main_output, benchmark_dict
         else:
-            return self.preprocessor.tokenizer.decode(outputs)
+            if stream:
+                return outputs
+            elif return_as_token_ids:
+                return outputs
+            else:
+                return self.preprocessor.tokenizer.decode(outputs)
```

Comment:
```suggestion
if not (stream or return_as_token_ids):
    outputs = self.preprocessor.tokenizer.decode(outputs)

if not benchmark:
    return outputs

benchmark_dict["Seconds total"] = time.perf_counter() - t0
benchmark_dict["Tokens generated"] = outputs.size(0) - input_ids.size(0)
benchmark_dict["Inference speed in tokens/sec"] = benchmark_dict["Tokens generated"] / benchmark_dict["Seconds total"]
if self.fabric.device.type == "cuda":
    benchmark_dict["GPU memory used in GB"] = torch.cuda.max_memory_allocated() / 1e9

return outputs, benchmark_dict
```

Perhaps something like this?
P.S. I didn't run it :)